core/trino-main/src/main/java/io/trino/operator/join/unspilled/HashBuilderOperator.java

```diff
@@ -321,14 +323,13 @@ private void disposeLookupSourceIfRequested()
             return;
         }
 
-        index.clear();
```

Comment:
check state index is null?

Reply:
`close` set's it to null unconditionally (as well as `lookupSourceSupplier`).  Actually let me remove the `lookupSourceSupplier = null` as well